dicom-archive/updateMRI_Upload.pl

```diff
@@ -126,64 +145,55 @@
 ################################################################
 
 # fetch tarchiveLibraryDir from ConfigSettings in the database
-my $tarchiveLibraryDir = &NeuroDB::DBI::getConfigSetting(
-                            \$dbh,'tarchiveLibraryDir'
-                            );
+
+my $configOB = NeuroDB::objectBroker::ConfigOB->new(db => $db);
+my $tarchiveLibraryDir = $configOB->getTarchiveLibraryDir();
+
 # determine tarchive path stored in the database (without tarchiveLibraryDir)
 my $tarchive_path = $tarchive;
-$tarchive_path    =~ s/$tarchiveLibraryDir\/?//g;  
+$tarchive_path    =~ s/$tarchiveLibraryDir\/?//g;
 
-my $where         = " WHERE t.ArchiveLocation =?";
-
-if ($globArchiveLocation) {
-    $where = " WHERE t.ArchiveLocation LIKE ?";
-    $tarchive_path = basename($tarchive);
-}
+# Check if there is already an mri upload record for the tarchive
+my $mriUploadOB = NeuroDB::objectBroker::MriUploadOB->new(db => $db);
+my $resultRef = $mriUploadOB->getWithTarchive(
+    GET_COUNT, $tarchive_path, $globArchiveLocation
+);
 
-($query = <<QUERY) =~ s/\n/ /gm;
-SELECT 
-  COUNT(*) 
-FROM 
-  mri_upload m 
-JOIN 
-  tarchive t ON (t.TarchiveID=m.TarchiveID) 
-QUERY
-$query .= $where;
-$sth = $dbh->prepare($query);
-$sth->execute($tarchive_path);
-my $count = $sth->fetchrow_array;
-if($count>0) {
-   print "\n\tERROR: the tarchive is already uploaded \n\n"; 
+if($resultRef->[0]->[0] > 0) {
+   print "\n\tERROR: the tarchive is already uploaded \n\n";
    exit 6;
-} 
-
+}
 
 ################################################################
 #####get the tarchiveid from tarchive table#####################
 ################################################################
-($query = <<QUERY) =~ s/\n/ /gm;
-SELECT 
-  t.TarchiveID 
-FROM 
-  tarchive t 
-QUERY
-$query .= $where;
-$sth = $dbh->prepare($query);
-$sth->execute("%".$tarchive_path."%");
-my $tarchiveID = $sth->fetchrow_array;
 
+my $tarchiveOB = NeuroDB::objectBroker::TarchiveOB->new(db => $db);
+$resultRef = $tarchiveOB->getByTarchiveLocation(
+    ['TarchiveID'], $tarchive_path, $globArchiveLocation
+);
+
+if(@$resultRef != 1) {
+    die sprintf(
+        "Unexpected number of tarchive records with location %s found: %d\n",
+        $tarchive_path,
+        scalar(@$resultRef)
+    );
+}
+my $tarchiveID = $resultRef->[0]->[0];
 
 ################################################################
  #####populate the mri_upload columns with the correct values####
 ################################################################
-($query = <<QUERY) =~ s/\n/ /gm;
-INSERT INTO mri_upload 
-  (UploadedBy, UploadDate, TarchiveID, DecompressedLocation) 
-VALUES
-  (?, now(), ?, ?)
-QUERY
-my $mri_upload_insert = $dbh->prepare($query);
-$mri_upload_insert->execute($User,$tarchiveID,$source_location);
+
+$mriUploadOB->insert(
+    {
+      UploadedBy           => $User,
+      UploadDate           => DateTime->now()->strftime('%Y-%m-%d %H:%M:%S'),
```

Comment:
running this gave me this time:
![time](https://user-images.githubusercontent.com/15198379/37620595-13f91030-2b93-11e8-909c-27b6bd586768.png)

not sure why, it was ~4:12pm not 8:12pm

Reply:
The output shows that the date *was* 20:12 and not 4:12pm. Am I missing something?

Reply:
well, I inserted/uploaded it at 4:12pm, not sure why the time stamp shows up as 8:12pm, that was my question...
I don't think the `strftime()` did any of that, did it?

Reply:
Ah OK. There might be a time difference between the client (your PC/laptop) and the server.